### ISSUE TITLE
estimating a model of two outcomes

### ISSUE THREAD

**Issue by won12** [jmxpearson/rlfit#1]
Hello, 

First of all, I would appreciate for developing such a convenient toolbox. 
I have a question about the function of toolbox.  I used it to estimate the model of my behavioral data with reinforcement learning task and it worked for the standard reinforcement learning model. But I also have another model with two outcomes and wonder whether I can estimate the model with this toolbox. I have tested the code of Q model with two outcomes but it didn't work. Could you help me to estimate two outcome model with rlfit? do you have any example for that? 

Thank you in advance.

**Comment by jmxpearson** [comment]
Thanks!

I don't have an end-to-end example, but it shouldn't take many tweaks to get this working:
- As you noted, `Q_two_outcome` is the relevant `Qfun` to drop in [here](https://github.com/jmxpearson/rlfit/blob/master/rlfit.m#L1).
- As you can see [here](https://github.com/jmxpearson/rlfit/blob/master/Q_two_outcome.m#L16), `Q_two_outcome` assumes that the `outcome` variable has two columns in this case, one for each outcome type.
- The model is commented in terms of monetary and social outcomes, and [assumes](https://github.com/jmxpearson/rlfit/blob/master/Q_two_outcome.m#L4) that the two outcomes have separate learning rates but are combined in the proportion `beta(3)` to give total action value. That is

``` matlab
Q = beta(3)*Q_mon + (1-beta(3))*Q_soc;
```

So as long as you have an `outcome` with two columns and make sure the lower bound (`lb`) and upper bound (`nb`) variables allow for _three_ parameters (two learning rates plus relative outcome weighting), I think it should work. 

This is the model we used for [this paper](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0107621) and [this one](https://scholar.google.com/citations?view_op=view_citation&hl=en&user=4whjDosAAAAJ&cstart=20&pagesize=80&citation_for_view=4whjDosAAAAJ:iH-uZ7U-co4C), so it has worked. If you run into an error, please do post here.

**Comment by won12** [comment]
Thank you very much for your kind answer and also for the interesting papers!

I didn't change the size of lb and ub. So I can use the Q model two outcome but I have one more question. I have tested with the data set I created before testing with my real data. 

I just added randomly generated one outcome column to the original testdata to create the data set containing two outcomes. The beta of Q model with testdata was about 0.8 and in the two outcome Q model, I could get 0.7 for beta(1) and 0.01 for beta(2) which was from the outcome randomly generated. The beta(1) and beta(2) make sense but I couldn't understand the third beta value which was 0.0659. I think Beta(3) should be larger than this, since the Beta(1) was larger than Beta(2). 
I got these values as beta 0.0453   0.7447  0.0111  0.0659. 

I'm not sure I understood correctly about the beta values here. Could you help me this? 
I attached the test data file I used. 

[testdata2.xlsx](https://github.com/jmxpearson/rlfit/files/529231/testdata2.xlsx)

**Comment by jmxpearson** [comment]
So the betas are, in order:
- the softmax choice parameter
- the learning rate for Outcome 1
- the learning rate for Outcome 2
- the weight of Outcome 1 (in [0, 1]) for the first Q value in the effective Q value

You can definitely have an outcome with a high learning rate that is mostly ignored by the choice process. This could be fixed by adding priors in a Bayesian model, but is not constrained here.

Something I'm missing with your test data, though: Did you generate choices based on these Q values? The model estimates Q values based on choice behavior, so if the choices in your test data ignore Outcome 1, then you might get a very small `beta(3)`. That parameter is really about capturing which outcome is most driving behavior. If both of your Q values are randomly generated, it's not surprising that the model might learn to (mostly) ignore the one with a lower correlation.
